File: HotelManagementBackend/src/main/java/com/internship/hotelmanagementbackend/service/HotelService.java
package com.internship.hotelmanagementbackend.service;

import com.internship.hotelmanagementbackend.dto.HotelDto;
import com.internship.hotelmanagementbackend.model.Hotel;

import java.util.List;

public interface HotelService {
    List<HotelDto> getAllInSpecifiedRadius(double radius, double latitude, double longitude);
}

File: HotelManagementFrontend/src/app/services/hotel.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Hotel} from "../models/Hotel";

@Injectable({
    providedIn: 'root'
})
export class HotelService {
    readonly baseURL = "http://localhost:8888/hotels";

    constructor(private httpClient: HttpClient) {
    }

    public getAllInSpecifiedRadius(radius: number, latitude: number, longitude: number): Observable<Hotel[]> {
        const headers = {'Content-Type': 'application/json'};
        return this.httpClient.get<Hotel[]>(this.baseURL + "/" + radius + '/' + latitude + '/' + longitude, {headers});
    }
}
